wasm/README.md

```diff
@@ -0,0 +1,42 @@
+# How to optimize your binary size?
+
+Smartcontracts should be optimized for because they are stored directly on the blockchain. By
+following this list of optimization steps your binary's size can be reduced by an order of magnitude
+(e.g. from 1.1MB to 100KB):
+
+1. Create a `Cargo.toml` akin to the following template:
+
+```toml
+  [package]
+  name = "smartcontract"
+  version = "0.1.0"
+  edition = "2021"
+
+  [lib]
+  crate-type = ['cdylib']    # Crate has to be linked dynamically
+
+  [profile.release]
+  strip = "debug"     # Remove debugging info from the binary
+  panic = "abort"     # Panics are transcribed to Traps when compiling for wasm anyways
+  lto = true          # Link-time-optimization produces notable decrease in binary size
+  opt-level = "z"     # Optimize for size vs speed with "s"/"z"(removes vectorization)
+  codegen-units = 1   # Further reduces binary size but increases compilation time
+
+  [dependencies]
+  iroha_data_model = { git = "https://github.com/hyperledger/iroha/", branch = "iroha2", default-features = false }
```

Comment:
Do we still need `no_std` if we do both LTO and dynamically link the smartcontracts?

Reply:
yes, it seems so. I think that is because LTO happens after successful compilation of modules and it's not possible to compile `std` for `wasm32` target